app/src/main/java/com/github/steroidteam/todolist/view/DrawingFragment.java

```diff
@@ -46,40 +53,46 @@ public View onCreateView(
         super.onCreate(savedInstanceState);
         View root = inflater.inflate(R.layout.fragment_drawing, container, false);
 
-        canvasLayout = root.findViewById(R.id.drawSpace);
-        colorPicker = root.findViewById(R.id.colorPicker);
-        colorPickerWindow = root.findViewById(R.id.colorPickerWindow);
-        saturationBar = root.findViewById(R.id.saturationbar);
-        valueBar = root.findViewById(R.id.valuebar);
-
-        root.findViewById(R.id.drawing_first_button).setOnClickListener(this::firstButton);
-        root.findViewById(R.id.drawing_second_button).setOnClickListener(this::secondButton);
-        root.findViewById(R.id.drawing_third_button).setOnClickListener(this::thirdButton);
-        root.findViewById(R.id.drawing_fourth_button).setOnClickListener(this::fourthButton);
-        root.findViewById(R.id.erase_button).setOnClickListener(this::eraseButton);
-        root.findViewById(R.id.colorChoose).setOnClickListener(this::ColorPickerButton);
-        root.findViewById(R.id.cancelColor).setOnClickListener(this::cancelColorButton);
-        root.findViewById(R.id.applyColor).setOnClickListener(this::applyColorButton);
-        root.findViewById(R.id.backButton)
+        canvasLayout = root.findViewById(R.id.drawing_space);
+        colorPicker = root.findViewById(R.id.drawing_color_picker);
+        colorPickerWindow = root.findViewById(R.id.drawing_color_picker_layout);
+        saturationBar = root.findViewById(R.id.drawing_saturation_bar);
+        valueBar = root.findViewById(R.id.drawing_value_bar);
+
+        root.findViewById(R.id.drawing_first_btn).setOnClickListener(this::firstButton);
+        root.findViewById(R.id.drawing_second_btn).setOnClickListener(this::secondButton);
+        root.findViewById(R.id.drawing_third_btn).setOnClickListener(this::thirdButton);
+        root.findViewById(R.id.drawing_fourth_btn).setOnClickListener(this::fourthButton);
+        root.findViewById(R.id.drawing_erase_btn).setOnClickListener(this::eraseButton);
+        root.findViewById(R.id.drawing_fifth_btn).setOnClickListener(this::ColorPickerButton);
+        root.findViewById(R.id.drawing_cancel_color_btn)
+                .setOnClickListener(this::cancelColorButton);
+        root.findViewById(R.id.drawing_apply_color_btn).setOnClickListener(this::applyColorButton);
+        root.findViewById(R.id.drawing_back_btn)
                 .setOnClickListener((view) -> getParentFragmentManager().popBackStack());
+        root.findViewById(R.id.drawing_save_btn).setOnClickListener(this::saveButton);
 
         drawingCanvas = new DrawingView(getContext());
         canvasLayout.addView(drawingCanvas);
         colorPicker.addSaturationBar(saturationBar);
         colorPicker.addValueBar(valueBar);
 
         setUpButtonAndColor(root);
+        // if (ContextCompat.checkSelfPermission(getContext(), android.permission.) !=
+        // PackageManager.PERMISSION_GRANTED){
+
+        // }
```

Comment:
I guess we can remove this?

Reply:
Thank you very much ! Absolutely, I should have removed it but I missed it 👍